### ISSUE TITLE
Using setup.py develop

### ISSUE THREAD

**Issue by nipunbatra** [conda-forge/staged-recipes#2667]
Hi there,
Many thanks for the effort in easing packaging!

I was trying to create a recipe for [NILM_Metadata](https://github.com/nilmtk/nilm_metadata). It has a single dependency (PyYaml). However, it needs to be installed using python setup.py develop, owing to some [package specific reasons](https://github.com/nilmtk/nilm_metadata/issues/6).

I was wondering if the python setup.py develop is causing issues in creating recipe. Link to the [PR](https://github.com/conda-forge/staged-recipes/pull/2665). Is there a workaround?

**Comment by jakirkham** [comment]
**TL;DR** Installing in develop mode is likely to be problematic. This should not be used for a `conda` recipe. The result will be a broken package.

**Long answer:** The reason being that a development install means that an `*.egg-link` file, which points back to the source directory. As a result, the final package will not actually contain the source code, but only a reference to the location of the source code on the system it was built on. Given that `conda-build` create a temporary directory to contain the source code while building and then removes this directory immediately afterwards, this link will point to a non-existent location after the build completes. This is why the package will be broken.

**Follow-up:** Could you please briefly explain why a develop install is needed? Unfortunately I lack the time to read the linked issues in detail, but my impression is there is some data that needs to be copied over with the package. If that is the case, I would recommend looking into `setuptools` options to do [this]( http://setuptools.readthedocs.io/en/latest/setuptools.html#including-data-files ) and propose something along these lines to the developers of the code base.

**Comment by jakirkham** [comment]
Going to close this out. Though feel free to respond to the questions raised.

**Comment by nipunbatra** [comment]
Thanks a lot for the clarification. I'm happy for the issue to be closed.
